src/marshaller/HipieDDL.js

```diff
@@ -241,15 +241,19 @@
     TableMappings.prototype.doMapAll = function (data) {
         var retVal = SourceMappings.prototype.doMapAll.apply(this, arguments);
         if (retVal instanceof Array) {
-            //var columnsRHS = this.visualization.source.getColumnsRHS();
-            this.visualization.fields.forEach(function (field, idx) {
+            var columnsRHSIdx = this.visualization.source.getColumnsRHSIdx();
+            this.visualization.fields.forEach(function (field) {
                 var fieldType = (!field || !field.properties) ? "unknown" : hipieType2DBType(field.properties.type);
-                switch (fieldType) {
-                    case "dataset":
-                        retVal = retVal.map(function (row) {
-                            row.forEach(function (cell, cellIdx) {
-                                if (cellIdx === idx) {
-                                    cell = cell.Row || cell;
+                var colIdx = columnsRHSIdx[field.id];
+                if (isNaN(colIdx)) {
```

Comment:
In this case it should be either a number or undefined so should be fine (I might change it to test that explicitly)